core/block.ts

```diff
@@ -997,7 +997,7 @@ export class Block implements IASTNodeLocation {
    * @param url URL string for block help, or function that returns a URL.  Null
    *     for no help.
    */
-  setHelpUrl(url: string | Function) {
+  setHelpUrl(url: string | (() => string)) {
```

Comment:
This is a breaking change and probably not a desirable one.

I was thinking that the counterargument would be "if you supply a function with a different signature then it won't actually _work_", but having looked at where `.helpUrl` is used it's clear that an existing app might well supply a function that [returns `false` or `undefined` to indicate that no help window should be opened](https://github.com/google/blockly/blob/ec8f9c8589fd0ea849eba4ae80f59cb878a0f24a/core/block_svg.ts#L608).

Reply:
That is not documented behavior. I don't think we need to conform to it: https://developers.google.com/blockly/guides/create-custom-blocks/define-blocks#help_url:~:text=Using%20the%20JavaScript%20API%2C%20blocks%20can%20specify%20a%20function%2C%20instead%20of%20a%20static%20string%2C%20which%20returns%20a%20URL%20string%2C%20thus%20allowing%20for%20dynamic%20help.

Reply:
OK, agreed.  And functions still have the option to return the (falsy) empty string if so desired.